packages/jbrowse-cli/src/commands/add-assembly.test.ts

```diff
@@ -0,0 +1,850 @@
+        { encoding: 'utf8' },
+      )
+      expect(JSON.parse(contents)).toEqual({
+        ...defaultConfig,
+        assemblies: [
+          {
+            ...baseAssembly,
+            refNameColors: [
+              'red',
+              'orange',
+              'yellow',
+              'green',
+              'blue',
+              'purple',
+            ],
+            sequence: {
+              ...baseSequence,
+              adapter: { type: 'CustomAdapter' },
+            },
+          },
+        ],
+      })
+    })
+
+  setup
+    .do(async () => {
+      await fsPromises.writeFile('config.json', '{}')
+    })
+    .command(['add-assembly', '{"type":"CustomAdapter"}', '--name', 'simple'])
+    .it('can use an existing config file', async ctx => {
+      expect(await fsPromises.readdir(ctx.dir)).toContain('config.json')
+      const contents = await fsPromises.readFile(
+        path.join(ctx.dir, 'config.json'),
+        { encoding: 'utf8' },
+      )
+      expect(JSON.parse(contents)).toEqual({
+        ...defaultConfig,
+        assemblies: [
+          {
+            ...baseAssembly,
+            sequence: {
+              ...baseSequence,
+              adapter: { type: 'CustomAdapter' },
+            },
+          },
+        ],
+      })
+    })
+
+  setup
+    .do(async ctx => {
+      const simple2bit = path.join(
+        __dirname,
+        '..',
+        '..',
+        'test',
+        'data',
+        'simple.2bit',
+      )
+      await fsPromises.copyFile(
+        simple2bit,
+        path.join(ctx.dir, path.basename(simple2bit)),
+      )
+    })
+    .command(['add-assembly', 'simple.2bit'])
+    .command(['add-assembly', 'simple.2bit', '--overwrite'])
+    .it('can use --overwrite to replace an existing assembly', async ctx => {
+      expect(await fsPromises.readdir(ctx.dir)).toContain('config.json')
+      const contents = await fsPromises.readFile(
+        path.join(ctx.dir, 'config.json'),
+        { encoding: 'utf8' },
+      )
+      expect(JSON.parse(contents).assemblies.length).toBe(1)
+    })
+
+  setup
+    .do(async ctx => {
+      await fsPromises.mkdir('jbrowse')
+      await fsPromises.rename(
+        'manifest.json',
+        path.join('jbrowse', 'manifest.json'),
+      )
+      const simple2bit = path.join(
+        __dirname,
+        '..',
+        '..',
+        'test',
+        'data',
+        'simple.2bit',
+      )
+      await fsPromises.copyFile(
+        simple2bit,
+        path.join(ctx.dir, path.basename(simple2bit)),
+      )
+      process.chdir('jbrowse')
+    })
+    .mockConsoleLog()
+    .mockConsoleError()
+    .command(['add-assembly', path.join('..', 'simple.2bit')])
+    .it('warns when a data file is not in the JBrowse directory', async ctx => {
+      expect(ctx.consoleLog).not.toHaveBeenCalled()
+      expect(ctx.consoleError.mock.calls[0][0]).toContain(
+        'is not in the JBrowse directory',
+      )
+    })
+
+  setup
+    .nock('https://mysite.com', site =>
```

Comment:
I think the process of actually validating the URL is interesting but maybe not required, i'd make it optional and maybe disabled by default.

Reply:
It is optional (see `--skipCheck`), but I think having it enabled by default will help catch errors sooner.